flink-autoscaler-plugin-jdbc/src/main/java/org/apache/flink/autoscaler/jdbc/event/JdbcAutoScalerEventHandler.java

```diff
@@ -104,6 +135,49 @@ public void handleScalingEvent(
         }
     }
 
+    @Override
+    public void close() {
+        if (Objects.nonNull(scheduledEventHandlerCleaner)
+                && !scheduledEventHandlerCleaner.isShutdown()) {
+            scheduledEventHandlerCleaner.shutdownNow();
+        }
+    }
+
+    @VisibleForTesting
+    void cleanExpiredEvents() {
+        var date =
+                Timestamp.from(
+                        jdbcEventInteractor
+                                .getCurrentInstant()
+                                .minusMillis(eventHandlerTtl.toMillis()));
+        try {
+            Long minId = jdbcEventInteractor.queryMinEventIdByCreateTime(date);
```

Comment:
We could define a nested loops:

- The outer loop check `Long minId = jdbcEventInteractor.queryMinEventIdByCreateTime(date)`, and `minId != null`
- The inner loop check `deleted == batchSize`.

Reply:
pls https://github.com/apache/flink-kubernetes-operator/pull/865#discussion_r1728461523